src/controllers/netease/redirect.js

```diff
@@ -0,0 +1,51 @@
+// This module is intended to redirect the ncm path
+const winston = require('winston')
+const sdk = require('NeteaseCloudMusicApi')
+const Joi = require('joi')
+
+// validation schema
+const { ValidateParams } = require('../../utils/response')
+const schema = Joi.object({
+  id: Joi.number().integer().min(1).max(1000000000000000).required(), // the max number is intended to prevent the Number(int) overflow
+  br: Joi.number().valid(999000, 320000, 128000).default(320000) // flac, 320k, 128k
+})
+
+module.exports = async (ctx) => {
+  const params = Object.assign({}, ctx.params, ctx.query, ctx.request.body)
+  if (!await ValidateParams(params, schema, ctx)) { // validateParams
+    return
+  }
+  const result = await sdk.song_url({
+    id: ctx.params.id,
+    realIP: ctx.get('X-Real-IP'),
+    br: params.br ?? 320000
+  })
+  winston.verbose(result)
+  if (result.status !== 200) {
+    ctx.status = Number.parseInt(result.status)
+    ctx.body = {
+      status: result.status,
+      message: '上游错误',
+      data: result.body,
+      ts: Date.now()
+    }
+    return
+  } else if (
```

Comment:
unnecessary else if,

Reply:
the result body like below:

{
cookies: xxx,
status: xxx, // HTTP Status code(performed by request)
body: {
code: xxx // the API ret code
}
}

Reply:
🧐 any further suggestions?